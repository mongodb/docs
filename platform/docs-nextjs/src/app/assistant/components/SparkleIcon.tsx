'use client';

import { createGlyphComponent } from '@leafygreen-ui/icon';

export const SparkleIcon = createGlyphComponent('Sparkle', (props: React.SVGProps<SVGSVGElement>) => (
  <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
    <path
      d="M6.27413 2.89343L5.27579 5.88842C5.17568 6.18877 4.94 6.42445 4.63965 6.52456L1.64466 7.52289C1.18616 7.67573 1.18615 8.32427 1.64466 8.47711L4.63965 9.47544C4.94 9.57555 5.17568 9.81123 5.27579 10.1116L6.27413 13.1066C6.42696 13.5651 7.0755 13.5651 7.22834 13.1066L8.22667 10.1116C8.32679 9.81123 8.56247 9.57555 8.86281 9.47544L11.8578 8.47711C12.3163 8.32427 12.3163 7.67573 11.8578 7.52289L8.86281 6.52456C8.56247 6.42445 8.32679 6.18877 8.22667 5.88842L7.22834 2.89343C7.0755 2.43492 6.42696 2.43492 6.27413 2.89343Z"
      fill="url(#paint0_linear_7198_11304)"
    />
    <path
      d="M12.5477 1.17194L12.3166 1.8651C12.2165 2.16545 11.9808 2.40113 11.6805 2.50125L10.9873 2.7323C10.758 2.80872 10.758 3.13299 10.9873 3.20941L11.6805 3.44046C11.9808 3.54058 12.2165 3.77626 12.3166 4.0766L12.5477 4.76977C12.6241 4.99902 12.9483 4.99902 13.0248 4.76977L13.2558 4.0766C13.3559 3.77626 13.5916 3.54058 13.892 3.44046L14.5851 3.20941C14.8144 3.13299 14.8144 2.80872 14.5851 2.7323L13.892 2.50125C13.5916 2.40113 13.3559 2.16545 13.2558 1.8651L13.0248 1.17194C12.9483 0.942687 12.6241 0.942687 12.5477 1.17194Z"
      fill="url(#paint1_linear_7198_11304)"
    />
    <path
      d="M12.5477 11.2302L12.3166 11.9234C12.2165 12.2237 11.9808 12.4594 11.6805 12.5595L10.9873 12.7906C10.758 12.867 10.758 13.1913 10.9873 13.2677L11.6805 13.4988C11.9808 13.5989 12.2165 13.8346 12.3166 14.1349L12.5477 14.8281C12.6241 15.0573 12.9483 15.0573 13.0248 14.8281L13.2558 14.1349C13.3559 13.8346 13.5916 13.5989 13.892 13.4988L14.5851 13.2677C14.8144 13.1913 14.8144 12.867 14.5851 12.7906L13.892 12.5595C13.5916 12.4594 13.3559 12.2237 13.2558 11.9234L13.0248 11.2302C12.9483 11.001 12.6241 11.001 12.5477 11.2302Z"
      fill="url(#paint2_linear_7198_11304)"
    />
    <defs>
      <linearGradient
        id="paint0_linear_7198_11304"
        x1="3.00078"
        y1="3"
        x2="12.5008"
        y2="14"
        gradientUnits="userSpaceOnUse"
      >
        <stop offset="0.0931517" stopColor="#00ED64" />
        <stop offset="1" stopColor="#0498EC" />
      </linearGradient>
      <linearGradient
        id="paint1_linear_7198_11304"
        x1="3.00078"
        y1="3"
        x2="12.5008"
        y2="14"
        gradientUnits="userSpaceOnUse"
      >
        <stop offset="0.0931517" stopColor="#00ED64" />
        <stop offset="1" stopColor="#0498EC" />
      </linearGradient>
      <linearGradient
        id="paint2_linear_7198_11304"
        x1="3.00078"
        y1="3"
        x2="12.5008"
        y2="14"
        gradientUnits="userSpaceOnUse"
      >
        <stop offset="0.0931517" stopColor="#00ED64" />
        <stop offset="1" stopColor="#0498EC" />
      </linearGradient>
    </defs>
  </svg>
));
