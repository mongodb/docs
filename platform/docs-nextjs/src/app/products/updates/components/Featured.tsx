'use client';
import { css, cx } from '@leafygreen-ui/emotion';
import { H3 } from '@leafygreen-ui/typography';
import { Body } from '@leafygreen-ui/typography';
import Card from '@leafygreen-ui/card';
import { palette } from '@leafygreen-ui/palette';
import { Chip } from '@leafygreen-ui/chip';
import type { ProductUpdateEntry } from '../services/contentstack';
import { useRouter } from 'next/navigation';
import { theme } from '@/styles/theme';

const featuredContainerStyle = css`
  max-width: 1440px;
  margin: 48px auto 0;
  padding: 0 24px;
  margin-bottom: 60px;

  @media ${theme.screenSize.mediumAndUp} {
    padding: 0 32px;
  }

  @media ${theme.screenSize.largeAndUp} {
    padding: 0 32px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    padding: 0 8px;
  }
`;

const featuredContentStyle = css`
  display: flex;
  flex-direction: row;
  gap: 16px;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin-top: 26px;

  @media ${theme.screenSize.mediumAndUp} {
    gap: 16px;
  }

  @media ${theme.screenSize.largeAndUp} {
    gap: 24px;
  }
`;

const featuredCardStyle = css`
  flex: 0 0 100%;
  min-width: 0;
  padding: 24px;
  min-height: 360px;

  @media ${theme.screenSize.mediumAndUp} {
    flex: 0 0 calc(50% - 8px);
    padding: 32px;
  }

  @media ${theme.screenSize.largeAndUp} {
    flex: 0 0 calc(33.333% - 16px);
    padding: 40px;
  }
`;

const dateStyle = css`
  font-size: 13px;
  line-height: 20px;
  color: ${palette.gray.dark2};
  margin-bottom: 16px;
`;

const titleStyle = css`
  font-size: 24px;
  line-height: 32px;
  margin-bottom: 8px;
  color: ${palette.black};
`;

const bodyStyle = css`
  font-size: 16px;
  line-height: 24px;
  color: ${palette.black};
  margin-bottom: 44px;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
`;

const chipsStyle = css`
  display: flex;
  flex-direction: row;
  gap: 8px;
  flex-wrap: wrap;
`;

const chipStyle = css`
  font-size: 16px;
  padding: 2px;
`;

const Featured = ({ updates }: { updates: ProductUpdateEntry[] }) => {
  const router = useRouter();

  // Filter and limit to 3 most recent featured updates
  const limitedFeaturedUpdates = updates
    .sort((a, b) => new Date(a.created_at).getTime() - new Date(b.created_at).getTime())
    .slice(0, 3);

  return (
    <div className={cx(featuredContainerStyle)}>
      <H3>Featured Updates</H3>
      <div className={cx(featuredContentStyle)}>
        {limitedFeaturedUpdates.map((update: ProductUpdateEntry) => {
          const createdAt = update.beamer_created_at || update.created_at;
          const date = new Date(createdAt).toLocaleDateString('en-US', {
            year: 'numeric',
            month: 'long',
            day: 'numeric',
          });

          const slug = update.title
            .toLowerCase()
            .replace(/[^a-z0-9]+/g, '-')
            .replace(/(^-|-$)/g, '');

          return (
            <Card
              key={update.uid}
              className={cx(featuredCardStyle)}
              onClick={() => router.push(`/products/updates/${slug}`)}
            >
              <Body className={cx(dateStyle)}>{date}</Body>
              <H3 className={cx(titleStyle)}>{update.title}</H3>
              <Body className={cx(bodyStyle)}>{update.multi_line}</Body>
              <div className={cx(chipsStyle)}>
                {/* Display all tags from the three tag fields */}
                {[
                  ...(update.tags_category || []),
                  ...(update.tags_offerings || []),
                  ...(update.tags_product || []),
                ].map((tag: string) => (
                  <Chip key={tag} label={tag} className={cx(chipStyle)} />
                ))}
              </div>
            </Card>
          );
        })}
      </div>
    </div>
  );
};

export default Featured;
