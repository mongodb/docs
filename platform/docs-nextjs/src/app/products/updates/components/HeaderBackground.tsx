import { theme } from '@/styles/theme';
import { css, cx } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';
import Image from 'next/image';

const headerBackgroundStyle = css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: ${palette.green.light3};
  border-radius: 0 0 40px 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;

  /* Tablet breakpoint: between small and medium */
  @media ${theme.screenSize.mediumAndUp} {
    height: 230px;
    border-radius: 0 0 80px 80px;
  }

  /* Desktop breakpoint: larger than medium */
  @media ${theme.screenSize.largeAndUp} {
    height: 384px;
    border-radius: 0 0 80px 80px;
  }
`;

const leftImageStyle = css`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;

  @media ${theme.screenSize.mediumAndUp} {
    top: 10px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    top: 0;
  }
`;

const rightImageStyle = css`
  position: absolute;
  top: 0px;
  right: 0px;
  z-index: 1;

  @media ${theme.screenSize.mediumAndUp} {
    top: 0px;
    right: -10px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    top: -20px;
    right: -20px;
  }
`;

const mobileImageStyle = css`
  display: block;

  @media ${theme.screenSize.mediumAndUp} {
    display: none;
  }
`;

const tabletImageStyle = css`
  display: none;

  @media ${theme.screenSize.mediumAndUp} {
    display: block;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    display: none;
  }
`;

const desktopImageStyle = css`
  display: none;

  @media ${theme.screenSize['2XLargeAndUp']} {
    display: block;
  }
`;

const HeaderBackground = () => {
  return (
    <div className={cx(headerBackgroundStyle)}>
      {/* Left image container */}
      <div className={cx(leftImageStyle)}>
        <Image
          src="/product-updates/header-mobile-left.svg"
          alt="header-left"
          width={220}
          height={220}
          className={cx(mobileImageStyle)}
        />
        <Image
          src="/product-updates/header-tablet-left.svg"
          alt="header-left"
          width={320}
          height={320}
          className={cx(tabletImageStyle)}
        />
        <Image
          src="/product-updates/header-desktop-left.svg"
          alt="header-left"
          width={300}
          height={300}
          className={cx(desktopImageStyle)}
        />
      </div>

      {/* Right image container */}
      <div className={cx(rightImageStyle)}>
        <Image
          src="/product-updates/header-mobile-right.svg"
          alt="header-right"
          width={160}
          height={160}
          className={cx(mobileImageStyle)}
        />
        <Image
          src="/product-updates/header-tablet-right.svg"
          alt="header-right"
          width={200}
          height={200}
          className={cx(tabletImageStyle)}
        />
        <Image
          src="/product-updates/header-desktop-right.svg"
          alt="header-right"
          width={360}
          height={360}
          className={cx(desktopImageStyle)}
        />
      </div>
    </div>
  );
};

export default HeaderBackground;
