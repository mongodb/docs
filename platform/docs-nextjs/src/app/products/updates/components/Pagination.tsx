import { css, cx } from '@leafygreen-ui/emotion';
import Icon from '@leafygreen-ui/icon';
import IconButton from '@leafygreen-ui/icon-button';
import { Option, Select } from '@leafygreen-ui/select';

interface PaginationProps {
  setCurrentPage: (page: number) => void;
  totalFilteredUpdates: number;
  currentPage: number;
}

export const Pagination = ({ totalFilteredUpdates, setCurrentPage, currentPage }: PaginationProps) => {
  const goBack = () => {
    const previousPage = currentPage - 1;
    setCurrentPage(previousPage);
  };

  const goNext = () => {
    const nextPage = currentPage + 1;
    setCurrentPage(nextPage);
  };

  return (
    <div
      className={cx(css`
        display: flex;
      `)}
    >
      <div
        className={cx(css`
          display: flex;
          align-items: center;
          gap: 8px;
        `)}
      >
        <Select
          aria-label="pagination"
          placeholder=""
          size="small"
          dropdownWidthBasis="option"
          value={String(currentPage)}
          allowDeselect={false}
          onChange={(value) => setCurrentPage(Number(value))}
        >
          {Array(totalFilteredUpdates)
            .fill(0)
            .map((_, i) => {
              const page = i + 1;
              return <Option key={`${page}`} value={`${page}`}>{`${page}`}</Option>;
            })}
        </Select>
        <span
          className={cx(css`
            font-size: 13px;
          `)}
        >
          of {totalFilteredUpdates}
        </span>
      </div>
      <div
        className={cx(css`
          display: flex;
          align-items: center;
          margin-left: 5px;
        `)}
      >
        <IconButton aria-label="previous" disabled={currentPage === 1} onClick={goBack}>
          <Icon glyph="ChevronLeft" />
        </IconButton>
        <IconButton aria-label="next" disabled={currentPage === totalFilteredUpdates} onClick={goNext}>
          <Icon glyph="ChevronRight" />
        </IconButton>
      </div>
    </div>
  );
};
