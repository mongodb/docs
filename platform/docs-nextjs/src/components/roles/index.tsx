export { default as RoleAbbr } from './abbr';
export { default as RoleClass } from './class';
export { default as RoleCommand } from './command';
export { default as RoleFile } from './file';
export { default as RoleGold } from './gold';
export { default as RoleGUILabel } from './gui-label';
export { default as RoleHighlight } from './highlight';
export { default as RoleIcon } from './icon';
export { default as RoleKbd } from './kbd';
export { default as RoleLinkNewTab } from './link-new-tab';
export { default as RoleManual } from './manual';
export { default as RoleRed } from './red';
export { default as RoleRequired } from './required';

export type { AbbrProps } from './abbr';
export type { RoleClassProps } from './class';
export type { RoleCommandProps } from './command';
export type { RoleFileProps } from './file';
export type { GoldProps } from './gold';
export type { RoleGUILabelProps } from './gui-label';
export type { HighlightProps } from './highlight';
export type { RoleIconProps } from './icon';
export type { KbdProps } from './kbd';
export type { LinkNewTabProps } from './link-new-tab';
export type { RoleManualProps } from './manual';
export type { RedProps } from './red';
