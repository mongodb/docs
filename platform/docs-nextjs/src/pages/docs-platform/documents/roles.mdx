# Roles

This page includes several examples of different roles used.
