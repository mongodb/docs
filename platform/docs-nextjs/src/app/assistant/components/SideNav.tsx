'use client';

import Icon from '@leafygreen-ui/icon';
import { SideNav, SideNavGroup, SideNavItem } from '@leafygreen-ui/side-nav';
import { css } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';
import { SparkleIcon } from './SparkleIcon';
import { theme } from '@/styles/theme';
import { useConversationContext, type Conversation } from '../contexts/ConversationContext';

const StyledDivider = () => (
  <hr
    className={css`
      margin: 6px 0;
      border: none;
      border-top: 1px solid ${palette.gray.light2};
    `}
  />
);

const styledTitle = css`
  font-weight: 600 !important;
  font-size: 13px;
  line-height: 20px;
  padding: 6.5px 16px !important;
  &:hover {
    background-color: unset !important;
  }
`;

const styledNewChat = css`
  .dark-theme & {
    color: ${palette.green.light2} !important;
  }
  color: ${palette.green.dark2} !important;
  font-weight: 600 !important;
  font-size: 12px !important;
  letter-spacing: 0.4px !important;
  text-transform: uppercase !important;
`;

const styledRecentChats = css`
  > div {
    color: ${palette.gray.dark1};

    .dark-theme & {
      color: ${palette.gray.light1};
    }
    padding: 14px;
  }

  svg {
    color: ${palette.gray.dark1};

    .dark-theme & {
      color: ${palette.gray.light1};
    }
  }
`;

const conversationWrapperStyle = css`
  position: relative;
`;

const sideNavContainerStyle = css`
  position: relative;
  z-index: 2;

  @media ${theme.screenSize.upToSmall} {
    position: absolute;
    height: 80vh;
    background: white;
    box-shadow: 2px 0 10px rgba(0, 0, 0, 0.1);
  }
`;

export default function AssistantSideNav() {
  const { conversations, activeConversation, setCreateNewChat, setActiveConversation, setChangedConversation } =
    useConversationContext();

  const changeConvo = (convo: Conversation) => {
    if (activeConversation?.id !== convo.id) {
      setActiveConversation(convo);
      setChangedConversation(convo.id || null);
    }
  };

  return (
    <SideNav
      widthOverride={256}
      baseFontSize={14}
      aria-label="Full page MongoDB Assistant"
      className={sideNavContainerStyle}
    >
      <SideNavItem glyph={<SparkleIcon />} className={styledTitle} as={undefined}>
        MongoDB Assistant
      </SideNavItem>
      <StyledDivider />
      <SideNavItem glyph={<Icon glyph="Plus" />} className={styledNewChat} onClick={() => setCreateNewChat(true)}>
        New Chat
      </SideNavItem>
      <StyledDivider />
      <SideNavGroup glyph={<Icon glyph="ClockWithArrow" />} header={'Recent Chats'} className={styledRecentChats}>
        {conversations.map((conversation) => (
          <div key={conversation.id} className={conversationWrapperStyle}>
            <SideNavItem active={activeConversation?.id === conversation.id} onClick={() => changeConvo(conversation)}>
              {conversation.title}
            </SideNavItem>
          </div>
        ))}
      </SideNavGroup>
    </SideNav>
  );
}
