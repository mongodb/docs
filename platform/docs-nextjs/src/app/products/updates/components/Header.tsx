'use client';
import { css, cx } from '@leafygreen-ui/emotion';
import { H2 } from '@leafygreen-ui/typography';
import { Body } from '@leafygreen-ui/typography';
// import Button from '@leafygreen-ui/button';
import { palette } from '@leafygreen-ui/palette';
import HeaderBackground from './HeaderBackground';
import { theme } from '@/styles/theme';

const headerStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  height: 283px;
  width: 100%;
  position: relative;

  @media ${theme.screenSize.mediumAndUp} {
    height: 230px;
  }

  @media ${theme.screenSize.largeAndUp} {
    height: 384px;
  }
`;

const headerContentStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 782px;
  z-index: 2;
`;

const titleStyle = css`
  font-size: 32px;
  line-height: 40px;
  color: ${palette.black};
  margin-bottom: 8px;
  max-width: 326px;
  text-align: center;

  @media ${theme.screenSize.largeAndUp} {
    font-size: 64px;
    line-height: 72px;
    max-width: 763px;
  }
`;

const bodyStyle = css`
  text-align: center;
  margin-bottom: 20px;
  font-size: 13px;
  line-height: 20px;
  color: ${palette.black};
  max-width: 326px;

  @media ${theme.screenSize.mediumAndUp} {
    max-width: 520px;
  }

  @media ${theme.screenSize.largeAndUp} {
    font-size: 16px;
    line-height: 28px;
    max-width: 763px;
  }
`;

// const buttonStyle = css`
//     font-size: 13px;
//     line-height: 20px;
//     padding: 4px 12px;

//     @media ${theme.screenSize.mediumAndUp} {
//         padding: 8px 12px;
//     }

//     @media ${theme.screenSize.largeAndUp} {
//         font-size: 18px;
//         line-height: 24px;
//         padding: 12px 16px;
//         height: 48px;
//     }
// `;

const Header = () => {
  return (
    <div className={cx(headerStyle)}>
      <HeaderBackground />
      <div className={cx(headerContentStyle)}>
        <H2 className={cx(titleStyle)}>What&apos;s New at MongoDB?</H2>
        <Body className={cx(bodyStyle)}>
          Check out the latest updates in MongoDB â€“ including improvements to the developer experience, expanded
          workload support, app modernization tools, and more.
        </Body>
        {/* <Button variant="baseGreen" className={cx(buttonStyle)}>Subscribe via RSS Feed</Button> */}
      </div>
    </div>
  );
};

export default Header;
