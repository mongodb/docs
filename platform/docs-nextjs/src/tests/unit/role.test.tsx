import { render } from '@testing-library/react';

import { RoleAbbr, RoleGUILabel, RoleFile } from '@/components/roles/';
import type { ASTNode, TextNode } from '@/types/ast';

import mockDataGUILabel from '@/tests/data/role-guilabel.test.json';
import mockDataFile from '@/tests/data/role-file.test.json';
import mockDataAbbr from '@/tests/data/role-abbr.test.json';

describe('GUI Label', () => {
  it('correctly renders a "guilabel" role', () => {
    const tree = render(<RoleGUILabel nodeChildren={mockDataGUILabel.children as ASTNode[]} />);
    expect(tree.asFragment()).toMatchSnapshot();
  });
});

describe('File', () => {
  it('correctly renders a "file" role', () => {
    const tree = render(<RoleFile nodeChildren={mockDataFile.children as ASTNode[]} />);
    expect(tree.asFragment()).toMatchSnapshot();
  });
});

describe('Abbr', () => {
  it('correctly renders a "Abbr" role', () => {
    const abbr = render(<RoleAbbr nodeChildren={mockDataAbbr.children as [TextNode]} />);
    expect(abbr.asFragment()).toMatchSnapshot();
  });

  describe('when passed a value with format "ABBR (Full Name Here)"', () => {
    const mockValidAbbr = mockDataAbbr;
    mockValidAbbr.children[0].value = 'Display (Extended)';

    it('parses and strips whitespace from the display value', () => {
      const abbr = render(<RoleAbbr nodeChildren={mockValidAbbr.children as [TextNode]} />);
      expect(abbr.getByText('Display')).toBeTruthy();
    });
  });
});
