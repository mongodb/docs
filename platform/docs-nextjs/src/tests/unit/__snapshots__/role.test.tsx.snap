// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Abbr correctly renders a "Abbr" role 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  border-radius: 2px;
  -webkit-text-decoration: underline dotted 2px;
  text-decoration: underline dotted 2px;
  text-underline-offset: 0.125em;
  text-decoration-color: #001E2B;
}

.emotion-0:hover a>*::after {
  height: 0;
}

.emotion-0:focus,
.emotion-0:focus-visible {
  outline-color: #0498EC;
  outline-offset: 3px;
  outline-style: solid;
  outline-width: 2px;
}

.emotion-0:hover,
.emotion-0:focus,
.emotion-0:focus-visible {
  text-decoration-color: #001E2B;
}

<span
    class="emotion-0"
    tabindex="0"
  >
    Display
  </span>
</DocumentFragment>
`;

exports[`File correctly renders a "file" role 1`] = `
<DocumentFragment>
  <code
    class="file docutils literal"
  >
    <span
      class="pre"
    >
      examples/treasury_yield/src/main/resources/yield_historical_in.json
    </span>
  </code>
</DocumentFragment>
`;

exports[`GUI Label correctly renders a "guilabel" role 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-style: normal;
  font-weight: 700;
}

<span
    class="guilabel emotion-0"
  >
    Yes
  </span>
</DocumentFragment>
`;
